miio/airqualitymonitor.py

```diff
@@ -230,9 +267,20 @@ def __init__(
     def status(self) -> AirQualityMonitorStatus:
         """Return device status."""
 
+        if self.model is None:
+            """Autodetection"""
+            info = self.info()
+            self.model = info.model
+            self.mac_address = info.mac_address
+            self.firmware_version = info.firmware_version
+            self.hardware_version = info.hardware_version
```

Comment:
I would avoid exposing the mac, and the versions like this. Prefer `self._firmware_version` etc for internal use.